src/rust/engine/src/intrinsics.rs

```diff
@@ -197,25 +198,26 @@ fn multi_platform_process_request_to_process_result(
     })?;
 
     let platform_name: String = result.platform.into();
-    let gil = Python::acquire_gil();
-    let py = gil.python();
-    Ok(externs::unsafe_call(
-      py,
-      context.core.types.process_result,
-      &[
-        externs::store_bytes(py, &stdout_bytes),
-        Snapshot::store_file_digest(py, &context.core.types, &result.stdout_digest),
-        externs::store_bytes(py, &stderr_bytes),
-        Snapshot::store_file_digest(py, &context.core.types, &result.stderr_digest),
-        externs::store_i64(py, result.exit_code.into()),
-        Snapshot::store_directory_digest(py, &result.output_directory).map_err(|s| throw(&s))?,
-        externs::unsafe_call(
-          py,
-          context.core.types.platform,
-          &[externs::store_utf8(py, &platform_name)],
-        ),
-      ],
-    ))
+    let res = Python::with_gil(|py| {
```

Comment:
Continuing to use https://pyo3.rs/main/doc/pyo3/struct.Python.html#method.acquire_gil seems like it would avoid a bunch of nesting... it would also likely make your life easier to continue to use that strategy for now, to avoid introducing a bunch of closures.

Reply:
Yea, I think so.

Reply:
Or maybe I do it selectively? There are lots of cases where it's simpler to use `::with_gil()`, like going from 2 lines to 1. But then this call site is a good example of where it makes the diff 20 lines longer than it needs to be. 

So, prefer `with_gil`, but if it's going to make the diff a lot harder to read, stick with `acquire_gil` for now?

Reply:
Fine with me.